### PR TITLE
feat(queue): add canMakeFriends queue slot flag

### PR DESCRIPTION
**Pull request by garrappachc**

### CODE REVIEW

File: src/queue/queue-slot.ts [hunk 1]
```diff
@@ -5,4 +5,5 @@ export interface QueueSlot {
   gameClass: Tf2ClassName;
   playerId: string | null;
   ready: boolean;
+  canMakeFriends: boolean;
```
Comment (c43721) [inline src/queue/queue-slot.ts]: why not make this nullable? Then you don't need half the changes in this PR

you could then default this to false

**Review by c43721 — CHANGES_REQUESTED** [review]
One minor suggestion before moving on

File: src/queue/queue-slot.ts (hunk 1 above)
Comment (garrappachc) [reply src/queue/queue-slot.ts]: `canMakeFriendsWith` is an optional property now.

**Review by c43721 — APPROVED** [review]
Looks good now that we have nullability :)

**Pull request merged**